app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt

```diff
@@ -1040,6 +1036,119 @@ class PlayerFragment : BaseFragment(), OnlinePlayerOptions {
+                                        // seek to the time
+                                        exoPlayer.seekTo(timeInSeconds)
+                                    }
+                                }
+                                // youtube link without time
+                                // open new player
+                                playNextVideo(videoId)
+                            } else {
+                                // not the current video
+                                // open new player
+                                playNextVideo(videoId)
+                            }
+                        } else {
+                            // not a video link, might be channel or playlist link
+                            // handle normally
+                            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(link))
+                            startActivity(intent)
+                        }
+                    } else {
+                        // not a youtube link
+                        // handle normally
+                        val intent = Intent(Intent.ACTION_VIEW, Uri.parse(link))
+                        startActivity(intent)
+                    }
+                }
+            )
+        } else {
+            // Links can be present as plain text
+            descTextView.autoLinkMask = Linkify.WEB_URLS
+            descTextView.text = description
+        }
+    }
+
+    private fun getVideoIdIfVideoLink(link: String): String? {
+        if (link.contains("youtube.com")) {
+            // check if the link is a channel link
+            val videoId = link.substringAfter("v=").substringBefore("&")
+            if (videoId.isNotEmpty() && videoId != link) {
+                return videoId
+            }
+        } else if (link.contains("youtu.be")) {
+            val videoId = link.substringAfter("be/").substringBefore("&")
+            if (videoId.isNotEmpty() && videoId != link) {
+                return videoId
+            }
+        }
+        return null
+    }
+
+    private fun getTimeInMillis(time: String): Long {
+        var timeInSeconds = 0L
+        var found = false
+        // check if the time is in seconds
+        if (time.contains("s")) {
+            val longOrNull = time.substringBefore("s").toLongOrNull()
+            if (longOrNull != null) {
+                timeInSeconds += longOrNull
+                found = true
+            }
+        }
+        // check if the time is in minutes
+        if (time.contains("m")) {
+            val longOrNull = time.substringBefore("m").substringAfter("s").toLongOrNull()
+            if (longOrNull != null) {
+                timeInSeconds += longOrNull * 60
+                found = true
+            }
+        }
+        // check if the time is in hours
+        if (time.contains("h")) {
+            val longOrNull = time.substringBefore("h").substringAfter("m").toLongOrNull()
```

Comment:
Instead, please use something like ```kotlin
time.substringBefore("h").substringAfter("m").toLongOrNull()?.let {
    timeInSeconds += longOrNull * 60 * 60
    found = true
}
```
here

Reply:
Pardons but i thought this would be more readable.